UnityProject/Assets/Scripts/UI/Systems/Lobby/CharacterCustomization.cs

```diff
@@ -94,6 +94,12 @@ public class CharacterCustomization : MonoBehaviour
 
 		public InputField SerialiseData;
 
+		private BodyPartSpriteAndColour facialHairList;
```

Comment:
I feel like you should have a list of all the customisations and call randomise on those, because what happens when there is isn't facial hair for a species? 

it's fine for stuff like skin colour since that's built into the customisation screen itself

Reply:
problem with that comes in the form of this question : "how do we know what this should do with this customization here?"

It's much easier to create a "what if scenario" inside the code that lets us have a special set of rules for each race or customization. However, I will look into making this process dynamic as well later when I'm working on adding multiple characters in a later PR.

Reply:
I don't think currently there is any special code for a customisation currently, I think it's probably better to have the functionality be class-based, so let's say if moths had special wings for female and male,  you would inherit from thecolour and sprite customisation and add the custom logic there then you can reuse it on other stuff easily

Reply:
1. All Race data inherits from the same single class. We don't need to create any new classes for any races.
2. Randomization functionality inside the race class is absolutely terrible and creates more complexity for no reason. It's better to find a way to see what properties dynamic dropdowns have and have them dynamically change stuff without hardcoding their references like what I'm doing right now. (Technically not hardcoding but you get the point..)

Reply:
I'm not talking about inside the race class am talking about within the class that handles setting this Sprite colour and specified sprite, inheriting from BodyPartColourSprite : BodyPartCustomisationBase that is specified on each body part prefab

Reply:
so CharacterCustomization.cs? It sets the character's skin tone already there and then the game reads that skin tone and applies it to all sprites in the character customizer and in-game as well.
Generally speaking I'll work on adding a dynamic way to change dynamic customization options later, when that time comes we can talk about it on discord. but right now this works well for now and doesn't need any more changes to how we're handling this.

Reply:
no, in BodyPartCustomisationBase , you have a randomisation  function in there and you call that and randomise it itself

Reply:
![image](https://user-images.githubusercontent.com/34368774/112194746-e046c880-8c11-11eb-8407-4f971dad7b6d.png)
There is no such function like that in this class

Reply:
well I'm saying add the randomise function to this class  ,then you keep a list of BodyPartCustomisationBase  that are open currently, and when you press randomise you loop through that  and call random on each one

Reply:
Nope, won't work. And to get it to work will still introduce the complexity that I was talking about.